apps/android/app/src/main/java/chat/simplex/app/views/database/DatabaseErrorView.kt

```diff
@@ -39,88 +43,131 @@ fun DatabaseErrorView(
   var useKeychain by remember { mutableStateOf(appPreferences.storeDBPassphrase.get()) }
   val context = LocalContext.current
   val restoreDbFromBackup = remember { mutableStateOf(shouldShowRestoreDbButton(appPreferences, context)) }
-  val saveAndRunChatOnClick: () -> Unit = {
+
+  fun callRunChat(confirmMigrations: MigrationConfirmation? = null) {
+    val useKey = if (useKeychain) null else dbKey.value
+    runChat(useKey, confirmMigrations, chatDbStatus, progressIndicator, appPreferences)
+  }
+
+  fun saveAndRunChatOnClick() {
     DatabaseUtils.setDatabaseKey(dbKey.value)
     storedDBKey = dbKey.value
     appPreferences.storeDBPassphrase.set(true)
     useKeychain = true
     appPreferences.initialRandomDBPassphrase.set(false)
-    runChat(dbKey.value, chatDbStatus, progressIndicator, appPreferences)
+    callRunChat()
+  }
+
+  @Composable
+  fun DatabaseErrorDetails(@StringRes title: Int, content: @Composable ColumnScope.() -> Unit) {
+    Text(
+      generalGetString(title),
+      Modifier.padding(start = 16.dp, top = 16.dp, bottom = 16.dp),
+      style = MaterialTheme.typography.h1
+    )
+    SectionView(null, padding = PaddingValues(horizontal = DEFAULT_PADDING, vertical = DEFAULT_PADDING_HALF), content)
+  }
+
+  @Composable
+  fun fileNameText(dbFile: String) {
+    Text(String.format(generalGetString(R.string.file_with_path), dbFile.split("/").lastOrNull() ?: dbFile))
   }
-  val title = when (chatDbStatus.value) {
-    is DBMigrationResult.OK -> ""
-    is DBMigrationResult.ErrorNotADatabase -> if (useKeychain && !storedDBKey.isNullOrEmpty())
-      generalGetString(R.string.wrong_passphrase)
-    else
-      generalGetString(R.string.encrypted_database)
-    is DBMigrationResult.Error -> generalGetString(R.string.database_error)
-    is DBMigrationResult.ErrorKeychain -> generalGetString(R.string.keychain_error)
-    is DBMigrationResult.Unknown -> generalGetString(R.string.database_error)
-    null -> "" // should never be here
+
+  @Composable
+  fun migrationsText(ms: List<String>) {
```

Comment:
Composable functions should be started with Uppercase letter (if they not return anything, e.g. non-unit)